sdk/nodejs/cmd/pulumi-language-nodejs/main_test.go

```diff
@@ -369,6 +369,296 @@ func TestGetRequiredPluginsNestedPolicyPack(t *testing.T) {
 	}, actual)
 }
 
+type filePathAndContents struct {
+	path    string
+	content string
+}
+
+func setupFiles(t *testing.T, files []filePathAndContents) string {
+	dir := filepath.Join(t.TempDir(), "program-dependency-testdir")
+	err := os.Mkdir(dir, 0o755)
+	require.NoError(t, err)
+
+	for i := range files {
+		files[i].path = filepath.Join(dir, files[i].path)
+	}
+
+	for _, file := range files {
+		err := os.MkdirAll(filepath.Dir(file.path), 0o755)
+		require.NoError(t, err)
+		err = os.WriteFile(file.path, []byte(file.content), 0o600)
+		require.NoError(t, err)
+	}
+	return dir
+}
+
+func readFileToString(filePath string) string {
+	fileBytes, err := os.ReadFile(filePath)
+	if err != nil {
+		panic(err)
+	}
+	return string(fileBytes)
+}
+
+func TestGetProgramDependencies(t *testing.T) {
+	t.Parallel()
+
+	t.Run("no-package.json-no-lock-files", func(t *testing.T) {
+		t.Parallel()
+
+		files := []filePathAndContents{
+			{
+				path:    "Pulumi.yaml",
+				content: `name: test`,
+			},
+		}
+
+		testDir := setupFiles(t, files)
```

Comment:
I'd be tempted to not define `file`s and just inline the construction into `setupFiles`